homeassistant/components/aftership/strings.json

```diff
@@ -0,0 +1,23 @@
+{
+  "title": "AfterShip",
+  "config": {
+    "step": {
+      "user": {
+        "title": "Setup AfterShip",
+        "description": "Grab your [AfterShip API key](https://admin.aftership.com/settings/api-keys) to authenticate with Home Assistant.",
+        "data": {
+          "name": "Integration name",
+          "api_key": "[%key:common::config_flow::data::api_key%]"
+        }
+      }
+    },
+    "error": {
+      "cannot_connect": "Failed to connect, please try again",
+      "invalid_auth": "Invalid authentication, check your API key",
+      "unknown": "Unexpected error"
+    },
+    "abort": {
+      "already_configured": "This AfterShip API key is already configured"
```

Comment:
This is never used in config flow.

Reply:
Ok so there must be something I'm doing wrong regarding this already_configured abort. To me this message should be displayed in the form when the user tries to add a already setup API key.

This is why I set the unique_id using the API key, because the Aftership API doesn't provide anything useful (like a username, mail...). I understood that in the config flow step, `self._abort_if_unique_id_configured()` should abort the flow and inform the user in the form with the `already_configured` string. But this is not what's happening, the form shows " Unexpected error" and there is an exception in the logs :

```
<class 'Exception'>
2020-05-16 21:02:59 ERROR (MainThread) [homeassistant.components.aftership.config_flow] Unexpected exception
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/components/aftership/config_flow.py", line 75, in async_step_user
    self._abort_if_unique_id_configured()
  File "/workspaces/core/homeassistant/config_entries.py", line 840, in _abort_if_unique_id_configured
    raise data_entry_flow.AbortFlow("already_configured")
homeassistant.data_entry_flow.AbortFlow: Flow aborted: already_configured
```

Should I simply drop the already configured "feature" of the config flow ?

Reply:
To show `already_configured` string to the user, your step should return `self.async_abort`. `self._abort_if_unique_id_configured()` is going to raise an `AbortFlow` exception which causes "Unexpected error". The standard way of doing this is manually check already configured entries.
https://github.com/home-assistant/core/blob/fca96799a0fb26d54c5e3d6b70528daf11d9a38c/homeassistant/components/songpal/config_flow.py#L76-L79
https://github.com/home-assistant/core/blob/fca96799a0fb26d54c5e3d6b70528daf11d9a38c/homeassistant/components/songpal/config_flow.py#L148-L153